src/utils/atomWithObservable.ts

```diff
@@ -13,41 +16,40 @@ type Subscription = {
 
 type Observer<T> = {
   next: (value: T) => void
-  error: (error: unknown) => void
+  error: (error: AnyError) => void
   complete: () => void
 }
 
 type ObservableLike<T> = {
   subscribe(observer: Observer<T>): Subscription
   subscribe(
     next: (value: T) => void,
-    error?: (error: unknown) => void,
+    error?: (error: AnyError) => void,
     complete?: () => void
   ): Subscription
   [Symbol.observable]?: () => ObservableLike<T> | undefined
 }
 
 type SubjectLike<T> = ObservableLike<T> & Observer<T>
 
-type InitialValueFunction<T> = () => T | undefined
-
-type AtomWithObservableOptions<TData> = {
-  initialValue?: TData | InitialValueFunction<TData>
+type Options<Data> = {
+  initialValue?: Data | (() => Data)
+  timeout?: number
```

Comment:
This new option is still an undocumented one. It only exists to avoid memory leaks with concurrent rendering. we are not sure what's the best practice. Hmm, maybe we should add `unstable_` prefix.